contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -411,6 +459,26 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
         _rebalance(_amountToSwap, _swapWeth, _minTokenReceived);
     }
 
+    function rebalanceWithVolumeLimit(
+        uint256 _amountToSwap,
+        bool _swapWeth,
+        uint256 _minTokenReceived
+    ) external nonReentrant onlyRebalancer {
+        uint256 volumeRebalancedCurrentPeriod = _amountToSwap;
+
+        // no rebalance transaction from the rebalancer in the last hour
```

Comment:
We can probably change it so that we are doing it as regenerative rate limit instead. Let me think and get back to this in a bit

Reply:
I guess the implementation could be a linear decay of the tallied swap amounts. Meaning: 
- there is a certain threshold that musn't be surpassed
- the function keeps tally of the amount that has been swapped and each block decays (threshold / 900). (might make sense to make the decay a 3 hour thing instead of 1 to allow for some variance in the defender action)

Will implement this tomorrow and see if it can be done so it doesn't increase the complexity of the code too much

Reply:
Yeah, what I was thinking was basically something like `(currentTime - lastRebalanceTime) * regenerationAmountPerSecond`. We can probably have an upper limit as well (say 500 WETH per rebalance). This will ensure that you cannot rebalance more than 500 WETH per hour. This will also limit the operator/rebalancer from doing a rebalance back to back for any reason